src/svg/axis.js

```diff
@@ -0,0 +1,211 @@
+(function(d3, fc) {
+    'use strict';
+
+    // A drop-in replacement for the D3 axis, supporting the decorate pattern.
+    fc.svg.axis = function() {
+
+        var scale = d3.scale.identity(),
+              decorate = fc.util.fn.noop,
+              orient = 'bottom',
+              tickArguments = [10],
+              tickValues = null,
+              tickFormat = null,
+              outerTickSize = 6,
+              innerTickSize = 6,
+              tickPadding = 3,
+              svgDomainLine = d3.svg.line();
+
+        var dataJoin = fc.util.dataJoin()
+            .selector('g.tick')
+            .element('g')
+            .key(fc.util.fn.identity)
+            .attrs({'class': 'tick'});
+
+        // returns a function that creates a translation based on
+        // the bound data
+        function containerTranslate(s, trans) {
+            return function(d) {
+                return trans(s(d), 0);
+            };
+        }
+
+        function translate(x, y) {
+            return 'translate(' + x + ', ' + y + ')';
+        }
+
+        function translateTransposed(x, y) {
+            return 'translate(' + y + ', ' + x + ')';
+        }
+
+        function transposeArray(arr) {
+            return arr.map(function(d) {
+                return [d[1], d[0]];
+            });
+        }
+
+        function isVertical() {
+            return orient === 'left' || orient === 'right';
+        }
+
+        var axis = function(selection) {
+
+            selection.each(function(data, index) {
+
+                // Stash a snapshot of the new scale, and retrieve the old snapshot.
+                var scaleOld = this.__chart__ || scale;
+                this.__chart__ = scale.copy();
+
+                function tryApply(fn, defaultVal) {
+                    return scale[fn] ? scale[fn].apply(scale, tickArguments) : defaultVal;
+                }
+
+                var ticksArray = tickValues == null ? tryApply('ticks', scale.domain()) : tickValues;
+                var tickFormatter = tickFormat == null ? tryApply('tickFormat', fc.util.fn.identity) : tickFormat;
+
+                // orientation logic
+                var sign = orient === 'bottom' || orient === 'right' ? 1 : -1;
+                var pathTranspose = isVertical() ? transposeArray : fc.util.fn.identity;
```

Comment:
Could this and the next line have the conditionals collapsed into the helper functions?